src/components/store/AddCartButton/index.tsx

```diff
@@ -0,0 +1,91 @@
+import { Dropdown } from '@/components/common';
+import { useEffect, useId, useState } from 'react';
+import styles from './style.module.scss';
+
+interface AddCartButtonProps {
+  currSize: string | undefined;
+  // Justification for disabling rules: This seems to be a false positive.
+  // https://stackoverflow.com/q/63767199/
+  // eslint-disable-next-line no-unused-vars
+  inStock: boolean;
+  inCart: boolean;
+  lifetimeRemaining: number;
+  montlyRemaining: number;
+  setInCart: (value: boolean) => void;
+}
+
+const AddCartButton = ({
+  currSize,
+  inStock,
+  inCart,
+  setInCart,
+  lifetimeRemaining,
+  montlyRemaining,
+}: AddCartButtonProps) => {
+  const [mounted, setMounted] = useState(false);
+  const [amount, setAmount] = useState<number>(1);
+
+  const myID = useId();
+
+  const maxCanBuy = Math.min(lifetimeRemaining, montlyRemaining);
+
+  useEffect(() => {
+    // console.log(`currSize ${currSize} inStock ${inStock} inCart ${inCart}`);
+    setMounted(true);
+  }, []);
+
+  if (!mounted) {
+    return null;
+  }
+
+  const validText = inStock ? (
+    <p className={styles.valid}>This item is in stock.</p>
+  ) : (
+    <p className={styles.error}>This item is out of stock.</p>
+  );
+
+  const optionArr: Array<{ value: string; label: string } | '---'> = [];
+
+  for (let i = 1; i <= maxCanBuy; i += 1) {
+    optionArr.push({ value: `${i}`, label: `${i}` });
+  }
+
+  optionArr.push('---');
+
+  return (
+    <div className={styles.addCartGroup}>
+      {maxCanBuy > 0 ? (
+        <p>You can buy up to {maxCanBuy} of this item.</p>
+      ) : (
+        <p>You can&apos;t buy any more of this item!.</p>
+      )}
+      <p>{`In cart: ${inCart}`}</p>
+      {currSize === undefined ? <p className={styles.error}>Please select a size.</p> : validText}
+      <div className={styles.buttonRow}>
+        <div className={styles.quantityColumn}>
+          <h4>Quantity</h4>
+          <Dropdown
+            name={`options${myID}`}
+            ariaLabel={`Dropdown to select the number of items to purchase for ${myID}`}
+            options={optionArr}
+            value={`${amount}`}
+            onChange={val => setAmount(Number(val))}
+          />
+        </div>
+
+        <input
+          className={`${inStock ? styles.buttonInStock : styles.buttonNoStock} ${styles.button}`}
+          type="button"
+          value={inStock ? 'Add to Cart' : 'Out of Stock'}
+          onClick={() => {
+            setInCart(!inCart);
+          }}
```

Comment:
also, the default border is showing on the button

![image](https://github.com/acmucsd/membership-portal-ui-v2/assets/22133785/88f856d6-0267-4237-8e72-cd9e26de6084)

we do remove the border by default, but only for `<button>` elements. I think you should change `<input type="button">` to `<button type="button">` for consistency with the rest of the codebase. it would fix the border

Reply:
![image](https://github.com/acmucsd/membership-portal-ui-v2/assets/54158686/f5c39345-b116-4722-b9c0-c3576b371431)
![image](https://github.com/acmucsd/membership-portal-ui-v2/assets/54158686/54c1c5fd-1f27-4f84-aa9b-c723e28cd165)

The Figma has a border on the button. Should I put the border back?

Without a border it looks like this:
![Uploading image.png…]()

Reply:
hmm, I think you should keep the border when the button is greyed out. I think the button looks better without the border when it's blue, but it's up to you whether the border on the blue button looks nicer